covid19_rsid_export.py

```diff
@@ -40,45 +40,49 @@ def _get_allele_str(alleles: str) -> str:
 
 
 def get_search_json_entry(release_record: dict) -> dict:
-    studies = set([ssInfo["study"] for ssInfo in release_record["ssInfo"]])
-    search_index_entry = {
-        "fields": [
-            {
-                "name": "id",
-                "value": f'rs{release_record["accession"]}'
-            },
-            {
-                "name": "chromosome",
-                "value": release_record["contig"]
-            },
-            {
-                "name": "start",
-                "value": release_record["start"]
-            },
-            {
-                "name": "variant_type",
-                "value": release_record["type"]
-            }
-        ],
-        "cross_references": [{"dbname": "ENA", "dbkey": study} for study in studies]
-    }
+    search_index_entries_per_allele = []
     for ssInfo in release_record["ssInfo"]:
-        search_index_entry["fields"].append({
-            "name": "alleles",
-            "value": f"Study: {ssInfo['study']}, "
-                     f"Reference/Alternate: "
-                     f"{_get_allele_str(ssInfo['ref'])}/{_get_allele_str(ssInfo['alt'])}"
-        })
+        search_index_entry = {
+            "fields": [
+                {
+                    "name": "id",
+                    "value": f'rs{release_record["accession"]}'
+                },
+                {
+                    "name": "chromosome",
+                    "value": release_record["contig"]
+                },
+                {
+                    "name": "start",
+                    "value": release_record["start"]
+                },
+                {
+                    "name": "variant_type",
+                    "value": release_record["type"]
+                },
+                {
+                    "name": "alleles",
+                    "value": f"Study: {ssInfo['study']}, "
+                             f"Reference/Alternate: "
+                             f"{_get_allele_str(ssInfo['refWithCtxBase'])}/{_get_allele_str(ssInfo['altWithCtxBase'])}"
```

Comment:
Why are we not putting the reference and alternate base in their own entry ?
I don't think they will be searchable in EBI search but it will make the json easier to read and I don't think it will cost us anything.
```python
{
    "name": "reference",
    "value": f"{_get_allele_str(ssInfo['refWithCtxBase'])}"
}
{
    "name": "alternate",
    "value": f"{_get_allele_str(ssInfo['altWithCtxBase'])}"
}
{
    "name": "studies",
    "value": f"{ssInfo['study']}"
}
```
Would this not work ?

Reply:
good point. in the new version where we have separate entries for ref/alt combination, this is more readable.